weave/graph_client.py

```diff
@@ -51,40 +76,67 @@ def run(self, run_id: str) -> typing.Optional[Run]:
             if run_attrs["span_id"] == None:
                 return None
             run_attrs = typing.cast(stream_data_interfaces.TraceSpanDict, run_attrs)
-            return Run(run_attrs)
+            return RunStreamTableSpan(run_attrs)
+
+    def find_op_run(
+        self, op_name: str, inputs: dict[str, typing.Any]
+    ) -> typing.Optional[RunStreamTableSpan]:
+        inputs_digest = hash_inputs(inputs)
+        from . import compile
+
+        with context_state.lazy_execution():
+            with compile.enable_compile():
+                rows_node = self.runs_st.rows()
+                filter_node = rows_node.filter(  # type: ignore
+                    lambda row: ops_primitives.Boolean.bool_and(
+                        row["name"] == op_name,
+                        row["attributes"]["_inputs_digest"] == inputs_digest,
+                    )
+                )[
+                    0
+                ]  # type: ignore
+                run_attrs = weave_internal.use(select_all(filter_node))
+                if not isinstance(run_attrs, dict):
+                    return None
+                if run_attrs.get("span_id") == None:
+                    return None
+                run_attrs = typing.cast(stream_data_interfaces.TraceSpanDict, run_attrs)
+                return RunStreamTableSpan(run_attrs)
 
-    def run_children(self, run_id: str) -> WeaveIter[Run]:
+    def run_children(self, run_id: str) -> WeaveIter[RunStreamTableSpan]:
```

Comment:
I like the use of id here as the lookup key. In other calls, we are using objects (like opDef or RunStreamTableSpan). I know this is evolving, but just want to callout that the id-style seems more portable.